### PR TITLE
Support the Ajax Spider

### LINKED ISSUE #223: Add Ajax Spider
I was waiting for the 'headless option' to become available, but theres actually no reason why it can be added now and then tweaked to support that options when its available.

### PR DESCRIPTION
**Pull request by psiinon**
Fixes #223
TODO

- [x] Get notifications when the ajax spider stops [#2047](https://github.com/zaproxy/zap-extensions/pull/2047)
- [x] Run browsers headless by default [#2055](https://github.com/zaproxy/zap-extensions/pull/2055)
- [x] Allow the browser to be chosen
- [x] Add to default tool list
- [x] Add to tutorial

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Updated to handle the ajax spider events

**Comment by psiinon** [comment]
Updated to allow the browser to be chosen and support headless ones.

**Comment by psiinon** [comment]
I think this is ready to be reviewed now.
It doesnt handle the case where the browsers dont get launched by ZAP, eg if they are not installed or the webdrivers are incompatible, but I think thats ok for now, and in any case I think other add-on will need to be changed in order to handle those cases effectively.

**Comment by psiinon** [comment]
Just updated the changelog ;)

File: src/main/zapHomeFiles/hudtutorial/en_GB/AjaxSpider.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	You will need to have these browsers installed in order to be able to use them. 
+	If you select the 'headless' option of either browser then they will be started in 'headless' mode so you will not see them running. 
+	This is usually the best option as this prevents new browsers from popping up on your screen. 
+	If you think the ajax spider is not handling your site correctly then try launching the non headless versions - 
+	that will allow you to see what they are doing.
+	<p>
+	When the ajax spider is started the tool will only show you that it is running - it does not currently show how far it has progressed. 
+	As with the traditional spider you will be able to see all of the pages that the ajax spider has found via the
+	Sites tree. If you click on the Ajax Spider tool while it is running then you will be given the option to stop it. 
+	You may well see alerts being raised after you start the ajax spider as ZAP passively scans all of the URLs it finds.
```
Comment (kingthorin) [inline src/main/zapHomeFiles/hudtutorial/en_GB/AjaxSpider.html]: I think this text should be consistent with "Ajax Spider" as a title/name.

Comment (psiinon) [reply src/main/zapHomeFiles/hudtutorial/en_GB/AjaxSpider.html]: I mostly copied the Spider page - that has 'spider' in lowercase except when talking about the 'Spider' tool, as here. Should we always capitalize the tool names everywhere in the tutorial?

Comment (kingthorin) [reply src/main/zapHomeFiles/hudtutorial/en_GB/AjaxSpider.html]: I dunno I just noticed that it wasn't consistent. Let's leave it for now.

Comment (kingthorin) [reply src/main/zapHomeFiles/hudtutorial/en_GB/AjaxSpider.html]: Resolving, as mentioned it's consistent with other similar content.

**Comment by thc202** [comment]
The build is failing (formatting issue).

**Comment by psiinon** [comment]
*sigh* I'll remember to spotlessApply one day...

**Pull request merged**